### PR TITLE
[NAE-1677] Message from exception thrown in SET event on data field is not propagated in EventOutcome

### PR DESCRIPTION
**Pull request by Coresto**
# Description

Refactoring of error handling so that error messages are displayed on the frontend

Fixes [NAE-1677]

## Dependencies

- No new dependencies were introduced

### Third party dependencies

- No new dependencies were introduced

### Blocking Pull requests

There are no dependencies on other PR

## How Has Been This Tested?

I created net with events on set and get in witch an exception was thrown and after that message from exception showed in snack bar as error.

### Test Configuration

| Name                | Tested on |
|---------------------| --------- |
| OS                  |   Windows 10        |
| Runtime             |  Java 11        |
| Dependency Manager  |  Maven 3.8.3         |
| Framework version   |  Spring Boot 2.7.0       |
| Run parameters      |           |
| Other configuration |           |

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] My changes have been checked, personally or remotely, with @machacjozef 
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have resolved all conflicts with the target branch of the PR
- [x] I have updated and synced my code with the target branch
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing tests pass locally with my changes:
    - [ ] Lint test
    - [ ] Unit tests
    - [ ] Integration tests
- [ ] I have checked my contribution with code analysis tools:
    - [ ] [SonarCloud](https://sonarcloud.io/project/overview?id=netgrif_application-engine)
    - [x] [Snyk](https://app.snyk.io/org/netgrif/project/ea82b3b8-658d-41f5-89a7-76cd600da01f)
- [ ] I have made corresponding changes to the documentation:
    - [ ] Developer documentation
    - [ ] User Guides
    - [ ] Migration Guides

[NAE-1677]: https://netgrif.atlassian.net/browse/NAE-1677?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/main/java/com/netgrif/application/engine/workflow/web/AbstractTaskController.java [hunk 1]
```diff
@@ -190,17 +190,35 @@ public CountResponse count(SingleElasticTaskSearchRequestAsList query, MergeFilt
 
 
     public EntityModel<EventOutcomeWithMessage> getData(String taskId, Locale locale) {
+        try {
         GetDataGroupsEventOutcome outcome = dataService.getDataGroups(taskId, locale);
         return EventOutcomeWithMessageResource.successMessage("Get data groups successful",
                 LocalisedEventOutcomeFactory.from(outcome,locale));
+        } catch (Exception e) {
```
Comment (minop) [inline src/main/java/com/netgrif/application/engine/workflow/web/AbstractTaskController.java:197]: do we really want to catch everything? How was the error-handling done so far?

Comment (Coresto) [reply src/main/java/com/netgrif/application/engine/workflow/web/AbstractTaskController.java:197]: i unified error handling implementation with assign, finish, cancel and delegate

**Review by minop — DISMISSED** [review]

File: src/main/java/com/netgrif/application/engine/workflow/web/AbstractTaskController.java [hunk 2]
```diff
@@ -190,17 +190,35 @@ public CountResponse count(SingleElasticTaskSearchRequestAsList query, MergeFilt
...
+            log.error("Get data on task [" + taskId + "] failed: ", e);
+            if (e instanceof IllegalArgumentWithChangedFieldsException) {
```
Comment (machacjozef) [inline src/main/java/com/netgrif/application/engine/workflow/web/AbstractTaskController.java:199]: } catch ( IllegalArgumentWithChangedFieldsException e) {
.......
} catch (Exception e) {
......
}

File: src/main/java/com/netgrif/application/engine/workflow/web/AbstractTaskController.java [hunk 3]
```diff
@@ -190,17 +190,35 @@ public CountResponse count(SingleElasticTaskSearchRequestAsList query, MergeFilt
...
+    public EntityModel<EventOutcomeWithMessage> setData(String taskId, ObjectNode dataBody, Locale locale) {
+        try {
+            Map<String, SetDataEventOutcome> outcomes = new HashMap<>();
+            dataBody.fields().forEachRemaining(it -> outcomes.put(it.getKey(), dataService.setData(it.getKey(), it.getValue().deepCopy())));
+            SetDataEventOutcome mainOutcome = taskService.getMainOutcome(outcomes, taskId);
+            return EventOutcomeWithMessageResource.successMessage("Data field values have been sucessfully set",
+                    LocalisedEventOutcomeFactory.from(mainOutcome, LocaleContextHolder.getLocale()));
+        } catch (Exception e) {
+            log.error("Set data on task [" + taskId + "] failed: ", e);
+            if (e instanceof IllegalArgumentWithChangedFieldsException) {
```
Comment (machacjozef) [inline src/main/java/com/netgrif/application/engine/workflow/web/AbstractTaskController.java:216]: } catch ( IllegalArgumentWithChangedFieldsException e) {
.......
} catch (Exception e) {
......
}

File: src/main/java/com/netgrif/application/engine/workflow/web/responsebodies/LocalisedTaskResource.java [hunk 4]
```diff
@@ -36,7 +36,7 @@ private void buildLinks() {
         add(WebMvcLinkBuilder.linkTo(WebMvcLinkBuilder.methodOn(TaskController.class)
                 .getData(task.getStringId(), null)).withRel("data"));
         add(WebMvcLinkBuilder.linkTo(WebMvcLinkBuilder.methodOn(TaskController.class)
-                .setData(null, task.getStringId(), null)).withRel("data-edit"));
+                .setData(task.getStringId(), null, null)).withRel("data-edit"));
```
Comment (Kovy95) [inline src/main/java/com/netgrif/application/engine/workflow/web/responsebodies/LocalisedTaskResource.java:39]: is it correct?

Comment (mladoniczky) [reply src/main/java/com/netgrif/application/engine/workflow/web/responsebodies/LocalisedTaskResource.java:39]: This is signature of method in question `public EntityModel<EventOutcomeWithMessage>  setData(Authentication auth, @PathVariable("id") String taskId, @RequestBody ObjectNode dataBody)` . So the order of the arguments is not correct.

Comment (machacjozef) [reply src/main/java/com/netgrif/application/engine/workflow/web/responsebodies/LocalisedTaskResource.java:39]: EntityModel<EventOutcomeWithMessage> setData(String taskId, ObjectNode dataBody, Locale locale)  its  okey...

**Review by mladoniczky — APPROVED** [review]

File: src/main/java/com/netgrif/application/engine/workflow/web/AbstractTaskController.java (hunk 3 above)
Comment (machacjozef) [reply src/main/java/com/netgrif/application/engine/workflow/web/AbstractTaskController.java]: Done

File: src/main/java/com/netgrif/application/engine/workflow/web/AbstractTaskController.java (hunk 2 above)
Comment (machacjozef) [reply src/main/java/com/netgrif/application/engine/workflow/web/AbstractTaskController.java]: Done

**Review by machacjozef — APPROVED** [review]

**Pull request merged**